src/libraries/Microsoft.Extensions.Logging.Console/src/WindowsLogConsole.cs

```diff
@@ -17,42 +17,108 @@ public WindowsLogConsole(bool stdErr = false)
         }
 
         private bool SetColor(ConsoleColor? background, ConsoleColor? foreground)
+        {
+            var backgroundChanged = SetBackgroundColor(background);
+            return SetForegroundColor(foreground) || backgroundChanged;
+        }
+
+        public bool SetBackgroundColor(ConsoleColor? background)
         {
             if (background.HasValue)
             {
                 System.Console.BackgroundColor = background.Value;
+                return true;
             }
+            return false;
+        }
 
+        public bool SetForegroundColor(ConsoleColor? foreground)
+        {
             if (foreground.HasValue)
             {
                 System.Console.ForegroundColor = foreground.Value;
+                return true;
             }
-
-            return background.HasValue || foreground.HasValue;
+            return false;
         }
 
         private void ResetColor()
         {
             System.Console.ResetColor();
         }
 
-        public void Write(string message, ConsoleColor? background, ConsoleColor? foreground)
+        public void Write(ReadOnlySpan<char> span, ConsoleColor? background, ConsoleColor? foreground)
         {
-            bool colorChanged = SetColor(background, foreground);
-            _textWriter.Write(message);
+            var colorChanged = SetColor(background, foreground);
+            _textWriter.Write(span);
             if (colorChanged)
             {
                 ResetColor();
             }
         }
 
-        public void WriteLine(string message, ConsoleColor? background, ConsoleColor? foreground)
+        public void Write(string message)
         {
-            bool colorChanged = SetColor(background, foreground);
-            _textWriter.WriteLine(message);
-            if (colorChanged)
+            // writes content with the embedded foreground and background ansi colored. it skips non-color related ansi codes
+            (int startIndex, int length, ConsoleColor? bg, ConsoleColor? fg) content = (-1, 0, null, null);
+            var span = message.AsSpan();
+            const char EscapeChar = '\x1B';
```

Comment:
could you please add a comment telling what we are parsing? something mentioning:

Set Display Attributes
Set Attribute Mode <ESC>[{attr1};...;{attrn}m
Sets multiple display attribute settings. The following lists standard attributes:
0 Reset all attributes
1 Bright
2 Dim
4 Underscore 
5 Blink
7 Reverse
8 Hidden

 Foreground Colours
30 Black
31 Red
32 Green
33 Yellow
34 Blue
35 Magenta
36 Cyan
37 White

 Background Colours
40 Black
41 Red
42 Green
43 Yellow
44 Blue
45 Magenta
46 Cyan
47 White